karmada search panic when create a new ResourceRegistry object

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
-->

**What happened**:
create a new ResourceRegistry object, karmada search panic:
```
I0603 11:12:51.527260      26 defaultstore.go:16] create default backend store: member1
I0603 11:12:51.527339      26 defaultstore.go:16] create default backend store: member2
I0603 11:12:51.527369      26 defaultstore.go:16] create default backend store: member3
I0603 11:12:51.527407      26 controller.go:222] add informer for member1, /v1, Resource=pods
E0603 11:12:51.527616      26 runtime.go:78] Observed a panic: "comparing uncomparable type cache.ResourceEventHandlerFuncs" (runtime error: comparing uncomparable type cache.ResourceEventHandlerFuncs)
goroutine 1023 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x2087240, 0xc001139ad0})
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0x7d
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc001139930})
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x75
panic({0x2087240, 0xc001139ad0})
	/usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/karmada-io/karmada/pkg/util/informermanager.(*singleClusterInformerManagerImpl).IsHandlerExist(0x2, {{0x0, 0x0}, {0xc0008a8e68, 0x2}, {0xc0004a337c, 0x4}}, {0x2753660, 0xc000919e30})
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/pkg/util/informermanager/single-cluster-manager.go:116 +0x1c5
github.com/karmada-io/karmada/pkg/util/informermanager.(*singleClusterInformerManagerImpl).ForResource(0xc000a3fb00, {{0x0, 0x0}, {0xc0008a8e68, 0x2}, {0xc0004a337c, 0x4}}, {0x2753660, 0xc000919e30})
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/pkg/util/informermanager/single-cluster-manager.go:90 +0x72
github.com/karmada-io/karmada/pkg/search.(*Controller).doCacheCluster(0xc0006e2500, {0xc002b06e47, 0x7})
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/pkg/search/controller.go:223 +0x913
github.com/karmada-io/karmada/pkg/search.(*Controller).cacheNext(0xc0006e2500)
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/pkg/search/controller.go:147 +0x105
github.com/karmada-io/karmada/pkg/search.(*Controller).worker(...)
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/pkg/search/controller.go:129
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x7fda97c715c0)
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x67
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000754558, {0x2717a40, 0xc0010859e0}, 0x1, 0xc000690600)
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x10000b738, 0x3b9aca00, 0x0, 0xac, 0x43f365)
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x0, 0x0, 0x0)
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x25
created by github.com/karmada-io/karmada/pkg/search.(*Controller).Start
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/pkg/search/controller.go:118 +0x1b8
panic: runtime error: comparing uncomparable type cache.ResourceEventHandlerFuncs [recovered]
	panic: runtime error: comparing uncomparable type cache.ResourceEventHandlerFuncs

goroutine 1023 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc001139930})
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0xd8
panic({0x2087240, 0xc001139ad0})
	/usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/karmada-io/karmada/pkg/util/informermanager.(*singleClusterInformerManagerImpl).IsHandlerExist(0x2, {{0x0, 0x0}, {0xc0008a8e68, 0x2}, {0xc0004a337c, 0x4}}, {0x2753660, 0xc000919e30})
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/pkg/util/informermanager/single-cluster-manager.go:116 +0x1c5
github.com/karmada-io/karmada/pkg/util/informermanager.(*singleClusterInformerManagerImpl).ForResource(0xc000a3fb00, {{0x0, 0x0}, {0xc0008a8e68, 0x2}, {0xc0004a337c, 0x4}}, {0x2753660, 0xc000919e30})
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/pkg/util/informermanager/single-cluster-manager.go:90 +0x72
github.com/karmada-io/karmada/pkg/search.(*Controller).doCacheCluster(0xc0006e2500, {0xc002b06e47, 0x7})
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/pkg/search/controller.go:223 +0x913
github.com/karmada-io/karmada/pkg/search.(*Controller).cacheNext(0xc0006e2500)
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/pkg/search/controller.go:147 +0x105
github.com/karmada-io/karmada/pkg/search.(*Controller).worker(...)
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/pkg/search/controller.go:129
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x7fda97c715c0)
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x67
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000754558, {0x2717a40, 0xc0010859e0}, 0x1, 0xc000690600)
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x10000b738, 0x3b9aca00, 0x0, 0xac, 0x43f365)
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x0, 0x0, 0x0)
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x25
created by github.com/karmada-io/karmada/pkg/search.(*Controller).Start
	/Users/momo/go/src/git.wemomo.com/sa/octopus/karmada/pkg/search/controller.go:118 +0x1b8
```

**What you expected to happen**:
karmada search should not panic 

**How to reproduce it (as minimally and precisely as possible)**:
create  a new ResourceRegistry object
**Anything else we need to know?**:

**Environment**:
- Karmada version:
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
- Others:

--- a/pkg/search/backendstore/defaultstore.go
+++ b/pkg/search/backendstore/defaultstore.go
@@ -15,14 +15,15 @@ type Default struct {
 func NewDefaultBackend(cluster string) *Default {
 	klog.Infof("create default backend store: %s", cluster)
 	return &Default{
-		resourceEventHander: cache.ResourceEventHandlerFuncs{
+		resourceEventHander: &cache.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
 				us, ok := obj.(*unstructured.Unstructured)
 				if !ok {
 					klog.Errorf("unexpected type %T", obj)
 					return
 				}
-				klog.V(4).Infof("add %s %s", cluster, us.GroupVersionKind().String())
+				klog.V(4).Infof("AddFunc Cluster(%s) GVK(%s) Name(%s/%s)",
+					cluster, us.GroupVersionKind().String(), us.GetNamespace(), us.GetName())
 			},
 			UpdateFunc: func(oldObj, curObj interface{}) {
 				us, ok := curObj.(*unstructured.Unstructured)
@@ -30,7 +31,8 @@ func NewDefaultBackend(cluster string) *Default {
 					klog.Errorf("unexpected type %T", curObj)
 					return
 				}
-				klog.V(4).Infof("udpate %s %s", cluster, us.GroupVersionKind().String())
+				klog.V(4).Infof("UpdateFunc Cluster(%s) GVK(%s) Name(%s/%s)",
+					cluster, us.GroupVersionKind().String(), us.GetNamespace(), us.GetName())
 			},
 			DeleteFunc: func(obj interface{}) {
 				us, ok := obj.(*unstructured.Unstructured)
@@ -38,7 +40,8 @@ func NewDefaultBackend(cluster string) *Default {
 					klog.Errorf("unexpected type %T", obj)
 					return
 				}
-				klog.V(4).Infof("delete %s %s", cluster, us.GroupVersionKind().String())
+				klog.V(4).Infof("DeleteFunc Cluster(%s) GVK(%s) Name(%s/%s)",
+					cluster, us.GroupVersionKind().String(), us.GetNamespace(), us.GetName())
 			}}}
 }
 

